### PR TITLE
IDEMPIERE-6152 - Fix error when opening an Info Window with subquery on same table as main query

### PR DESCRIPTION
**Pull request by VitorVilla**
https://idempiere.atlassian.net/browse/IDEMPIERE-6152

* Adjustment for the use of aliases in the MLookupFactory class.

# Pull Request Checklist

- [ ] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [ ] My code follows the best practices of this project
- [ ] I have performed a self-review of my own code
- [ ] My code is easy to understand and review. 
- [ ] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [ ] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MLookupFactory.java [hunk 1]
```diff
@@ -599,25 +599,25 @@ static private MLookupInfo getLookup_Table (Properties ctx, Language language,
...
-		}
-		s_cacheRefTable.put(key.toString(), retValue.cloneIt());
-		return retValue;
-	}	//	getLookup_Table
-
+		}
+		s_cacheRefTable.put(key.toString(), retValue.cloneIt());
+		return retValue;
+	}	//	getLookup_Table
+
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MLookupFactory.java:620]: please remove all the white space only changes above.

File: org.adempiere.base/src/org/compiere/model/MLookupFactory.java [hunk 2]
```diff
@@ -1024,9 +1024,9 @@ static public String getLookup_TableDirEmbed (Language language,
 		//
 		StringBuilder embedSQL = new StringBuilder("SELECT ");
 
-		StringBuilder displayColumn = getDisplayColumn(language, TableName, list, BaseTable);
+		StringBuilder displayColumn = getDisplayColumn(language, "alias", list, BaseTable);
 		embedSQL.append(displayColumn.toString());
-		embedSQL.append(" FROM ").append(TableName);
+		embedSQL.append(" FROM ").append(TableName).append(" alias");
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MLookupFactory.java:1029]: Please check comment at:
https://idempiere.atlassian.net/browse/IDEMPIERE-6152?focusedCommentId=50872

Also, it's possible that "alias" is a [reserved word in some databases](https://www.drupal.org/docs/develop/coding-standards/list-of-sql-reserved-words), better to use a different name

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

**Comment by CarlosRuiz-globalqss** [comment]
@VitorVilla - four things about pull requests:

1 - we usually write the commit message with similar wording as the JIRA ticket - in this case I edited the subject of the pull request to make it more accurate

2 - as a second line of the commit message we usually add more explanation about the solution on the commit.  I added your original message on the pull request as part of the description

3 - we add a link to the ticket to ease navigation from peer reviewers

4 - you must fill the boxes of what you did on the "Pull Request Checklist"

File: org.adempiere.base/src/org/compiere/model/MLookupFactory.java (hunk 1 above)
Comment (hengsin) [reply org.adempiere.base/src/org/compiere/model/MLookupFactory.java:620]: Your new commit doesn't fix this. If you are using Eclipse, Turn off "Ignore white space" in Preferences > General > Compare/Patch.

Comment (hengsin) [reply org.adempiere.base/src/org/compiere/model/MLookupFactory.java:620]: Also, if you have install the any edit plugin, turn off "Remove trailing whitespace" in Preferences > General > Editors > AnyEdit Tools

**Comment by VitorVilla** [comment]
https://github.com/idempiere/idempiere/pull/2414 
New PR Opened, close in this to keep this discussion in history

**Pull request closed without merging**